func_test.go

```diff
@@ -199,6 +199,237 @@ func TestABI(t *testing.T) {
 	}
 }
 
+func TestABI_ArgumentPassing(t *testing.T) {
+	if runtime.GOOS == "windows" && runtime.GOARCH == "386" {
+		t.Skip("need a 32bit gcc to run this test") // TODO: find 32bit gcc for test
+	}
+	libFileName := filepath.Join(t.TempDir(), "abitest.so")
+	if err := buildSharedLib("CC", libFileName, filepath.Join("testdata", "abitest", "abi_test.c")); err != nil {
+		t.Fatal(err)
+	}
+	lib, err := load.OpenLibrary(libFileName)
+	if err != nil {
+		t.Fatalf("Dlopen(%q) failed: %v", libFileName, err)
+	}
+	t.Cleanup(func() {
+		if err := load.CloseLibrary(lib); err != nil {
+			t.Errorf("Failed to close library: %v", err)
+		}
+	})
+
+	tests := []struct {
+		name string
+		fn   interface{}
+		cFn  string
+		call func(interface{}) string
+		want string
+	}{
+		{
+			name: "10_int32_baseline",
+			fn:   new(func(*byte, uintptr, int32, int32, int32, int32, int32, int32, int32, int32, int32, int32)),
+			cFn:  "stack_10_int32",
+			call: func(f interface{}) string {
+				buf := make([]byte, 256)
+				(*f.(*func(*byte, uintptr, int32, int32, int32, int32, int32, int32, int32, int32, int32, int32)))(&buf[0], 256, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+				return string(buf[:strings.IndexByte(string(buf), 0)])
```

Comment:
Why not use [bytes.IndexByte](https://pkg.go.dev/bytes#IndexByte) to avoid a conversion to string?